internal/wal/wal_aof.go

```diff
@@ -18,43 +20,146 @@ import (
 	"google.golang.org/protobuf/proto"
 )
 
-var writeBuf bytes.Buffer
+const (
+	segmentPrefix = "segment-"
+)
+
+var (
+	bufferPool = sync.Pool{
+		New: func() interface{} {
+			return make([]byte, 0, 4096) // Pre-allocate 4KB buffer
+		},
+	}
+)
 
 type WALAOF struct {
-	file   *os.File
-	mutex  sync.Mutex
-	logDir string
+	mutex               sync.Mutex
+	syncMutex           sync.Mutex    // Separate mutex for syncing
+	file                *os.File      // Current WAL file
+	writer              io.Writer     // Writer interface, can be buffered or unbuffered
+	bufWriter           *bufio.Writer // Buffered writer if buffering is enabled
+	logDir              string
+	syncInterval        time.Duration // Interval for syncing data to disk
+	lastSyncTime        time.Time     // Last time data was synced
+	shouldFsync         bool
+	maxFileSize         int64 // Maximum size of a log segment file in bytes
+	maxSegments         int   // Maximum number of log segment files to keep
+	currentSegmentIndex int   // Index of the current log segment
+
+	rotationMode     string        // "time" or "size"
+	rotationInterval time.Duration // Time interval for rotation
+	rotationTimer    *time.Timer   // Timer for rotation
+	quit             chan struct{} // Channel to signal shutdown
+
+	isBuffered bool   // Determines if writes are buffered
+	writeMode  string // "fsync" or "default"
 }
 
-func NewAOFWAL(logDir string) (*WALAOF, error) {
-	return &WALAOF{
-		logDir: logDir,
-	}, nil
-}
+func NewAOFWAL(
+	logDir string,
+	enableFsync bool,
+	maxFileSize int64,
+	maxSegments int,
+	rotationMode string,
+	rotationInterval time.Duration,
+	isBuffered bool,
+	writeMode string,
+) (*WALAOF, error) {
+	if err := os.MkdirAll(logDir, os.ModePerm); err != nil {
+		return nil, fmt.Errorf("failed to create log directory: %w", err)
+	}
 
-func (w *WALAOF) Init(t time.Time) error {
-	slog.Debug("initializing WAL at", slog.Any("log-dir", w.logDir))
-	if err := os.MkdirAll(w.logDir, os.ModePerm); err != nil {
-		return fmt.Errorf("failed to create log directory: %w", err)
+	if rotationMode != "time" && rotationMode != "size" {
+		return nil, fmt.Errorf("invalid rotationMode: %s, must be 'time' or 'size'", rotationMode)
+	}
+
+	if rotationMode == "time" && rotationInterval <= 0 {
```

Comment:
rotationInterval should be `rotationIntervalSec` making it explicit that we are talking seconds here or milliseconds.